src/pathwayvis/components/map/map.component.ts

```diff
@@ -429,41 +453,40 @@ class MapComponentCtrl {
    * Loads context menu and fetches list of actions for selected map element
    */
   private _loadContextMenu(): void {
-    const selection = this._builder.selection;
     const contextMenu = d3.select('.map-context-menu');
-
-    d3.selectAll('.reaction, .reaction-label')
-      .style('cursor', 'pointer')
-      .on('contextmenu', (d) => {
-        this.contextElement = d;
-        this.contextActions = this.actions.getList({
-          type: 'map:reaction',
-          shared: this._mapOptions.getMapData(),
-          element: this.contextElement,
+    d3.selectAll('text.reaction-label').on('mouseenter', (d) => {
+      if (d3.select('div#tooltip-container').select('#knockoutbutton')) {
+        this._getContext();
+        d3.select('div#tooltip-container').select('#knockoutbutton').on('click', () => {
+          this.processActionClick(this.contextActions[0], this.contextElement, this);
+          if (this.contextElement) {
+            this.$scope.$apply();
+          }
         });
+        d3.select('div#tooltip-container').select('#objectivebutton').on('click', () => {
+          this.processActionClick(this.contextActions[1], this.contextElement, this);
+          if (this.contextElement) {
+            this.$scope.$apply();
+          }
+        });
+      }
+    });
+  }
 
-        if (this.contextElement) {
-          this._renderContextMenu(contextMenu);
-          (<MouseEvent> currentEvent).preventDefault();
-        }
-      });
 
-    d3.select(document).on('click', () => {
-      contextMenu.style('visibility', 'hidden');
+  private _getContext(): void {
+    let data = JSON.parse(d3.select('div#tooltip-container').select('#knockoutbutton').attr("data").
```

Comment:
Why `let`?

Reply:
I think it's better if you change the keys/values on the js object:

const {biggId, ...restData} = JSON.parse(d3.select('div#tooltip-container').select('#knockoutbutton').attr("data"))
const data = Object.assign({bigg_id: biggId}, ...restData)